### ISSUE TITLE
Race condition on document mark_as_read 

### ISSUE THREAD

**Issue by joofsh** [department-of-veterans-affairs/caseflow#3030]
We are having race conditions during the `DocumentView.find_or_create_by`. Maybe the user opens 2 tabs at the same time?

https://sentry.ds.va.gov/department-of-veterans-affairs/caseflow/issues/346/

# Dev AC
1) Verify we cannot get this error locally by doing the following:
- [ ] On master change the code in DocumentsController#mark_as_read to:       
```
      DocumentView.find_or_create_by(
        document_id: params[:id],
        user_id: current_user.id) do |t|
        sleep 3
        t.update!(first_viewed_at: Time.zone.now)
      end
      render json: {}
```
Delete all DocumentView records, then open a single document in a new tab twice in quick succession. The unique error should be thrown.
- [ ] On this branch, add the same sleep and open a single document in a new tab twice. Ensure the error is not thrown.

**Comment by mdbenjam** [comment]
Interesting. I'm thinking we should just catch this error and then run it again? @joofsh any thoughts of the right approach here?

**Comment by joofsh** [comment]
I think catch and re-run seems appropriate, assuming our analysis is correct and there isn't a deeper issue at hand 👍

**Comment by mdbenjam** [comment]
Fixed in #4029
